internal/operands/vm-console-proxy/reconcile_test.go

```diff
@@ -35,49 +38,153 @@ const (
 )
 
 var _ = Describe("VM Console Proxy Operand", func() {
-	var v *vmConsoleProxy
-	var mockedRequest *common.Request
+	const (
+		replicas int32 = 1
+	)
 
-	BeforeEach(func() {
-		v = getMockedVmConsoleProxyOperand()
-		mockedRequest = getMockedRequest()
-	})
+	var (
+		operand *vmConsoleProxy
+		request common.Request
+	)
 
-	It("should return new with test bundle correctly", func() {
-		vmConsoleProxy := New(getMockedTestBundle())
-		Expect(vmConsoleProxy.serviceAccount).ToNot(BeNil(), "service account should not be nil")
-		Expect(vmConsoleProxy.clusterRole).ToNot(BeNil(), "cluster role should not be nil")
-		Expect(vmConsoleProxy.clusterRoleBinding).ToNot(BeNil(), "cluster role binding should not be nil")
-		Expect(vmConsoleProxy.service).ToNot(BeNil(), "service should not be nil")
-		Expect(vmConsoleProxy.deployment).ToNot(BeNil(), "deployment should not be nil")
-		Expect(vmConsoleProxy.configMap).ToNot(BeNil(), "config map should not be nil")
+	BeforeEach(func() {
+		operand = New(getMockedTestBundle())
+		request = getMockedRequest()
 	})
 
 	It("should return name correctly", func() {
-		name := v.Name()
+		name := operand.Name()
 		Expect(name).To(Equal(operandName), "should return correct name")
 	})
 
 	It("should return functions from reconcile correclty", func() {
-		functions, err := v.Reconcile(mockedRequest)
+		functions, err := operand.Reconcile(&request)
 		Expect(err).ToNot(HaveOccurred(), "should not throw err")
 		Expect(len(functions)).To(Equal(7), "should return correct number of reconcile functions")
 	})
+
+	It("should create vm-console-proxy resources", func() {
+		_, err := operand.Reconcile(&request)
+		Expect(err).ToNot(HaveOccurred())
+
+		ExpectResourceExists(operand.serviceAccount, request)
+		ExpectResourceExists(operand.clusterRole, request)
+		ExpectResourceExists(operand.clusterRoleBinding, request)
+		ExpectResourceExists(operand.configMap, request)
+		ExpectResourceExists(operand.service, request)
+		ExpectResourceExists(operand.deployment, request)
```

Comment:
It's better if these tests check the resources from the mocked bundle, and don't access private fields of the operand. Because private fields can change, and unit tests should not depend on them.

Reply:
What do you mean by "resources from the mocked bundle", by loading it from `vm-console-proxy-bundle-test.yaml` or just a new function?

Reply:
By "mocked bundle", I mean the `*vm_console_proxy_bundle.Bundle` returned by `getMockedTestBundle()`.
So for example, in `BeforeEach()` you could store the bundle into a variable, and then here check that the resource from the bundle was created by the operand:
```go
ExpectResourceExists(bundle.Deployment, request)
```

This way the test does not access any private fields.

Reply:
@akrejcir Yes, got it. I think this is what I did initially but then changed it yesterday.

Reply:
Done.